CRM/Core/Payment/PayPalProIPN.php

```diff
@@ -457,7 +457,10 @@ public function main() {
       }
     }
 
-    $paymentProcessorID = self::getPayPalPaymentProcessorID();
+    $paymentProcessorID = CRM_Utils_Array::value('processor_id', $this->_inputParameters);
```

Comment:
I think it should be self::retrieve like the other places, it probably also should be in the getInput function (ie $input['payment_processor_id'] = self::retrieve('processor_id', 'Int', 'POST', FALSE) ?? self::getPayPalPaymentProcessorID();

And then pass input['payment_processor_id'] through  here

-